src/profiles.js

```diff
@@ -0,0 +1,191 @@
+import tv4 from 'tv4'
+import jsonlint from 'json-lint'
+import _ from 'lodash'
+import path from 'path'
+
+import Utils from './utils'
+
+const DEFAULT_REMOTE_PATH = 'http://schemas.datapackages.org/registry.csv'
+let DEFAULT_LOCAL_PATH
+if (!Utils.isBrowser) {
+  DEFAULT_LOCAL_PATH = path.join(__dirname, '..', 'schemas', 'registry.csv')
+}
+
+class Profiles {
+
+  constructor(remote = false) {
+    if (remote || Utils.isBrowser) {
+      this._registry = this._loadRegistry(DEFAULT_REMOTE_PATH)
+      this._base_path = this._loadBasePath(DEFAULT_REMOTE_PATH)
+    } else {
+      this._registry = this._loadRegistry(DEFAULT_LOCAL_PATH)
+      this._base_path = this._loadBasePath(DEFAULT_LOCAL_PATH)
+    }
+  }
+
+  // ------ Profiles methods -------
+
+  /**
+   * Retrieve a profile by id.
+   *
+   * @param profile
+   * @return {Promise}
+   */
+  retrieve(profile) {
+    return this._getProfile(profile)
+  }
+
+  /**
+   * Validate a descriptor against a profile. You can provide custom schema
+   * or specify the profile id.
+   *
+   * @param {String|JSON} descriptor The descriptor that needs to be validated
+   * @param {String|JSON} profile Schema to validate against
+   * @return {Promise} Resolves `true` or array of strings which explain the errors.
+   */
+  validate(descriptor, profile = 'base') {
+    let json = descriptor
+
+    if (typeof descriptor === 'string') {
+      const lint = jsonlint(descriptor)
+
+      if (lint.error) {
+        return new Promise((resolve, reject) => {
+          reject(Error(lint.error.toString()))
+        })
+      }
+
+      json = JSON.parse(descriptor)
+    }
+
+    if (_.isObject(profile) && !_.isArray(profile) && !_.isFunction(profile)) {
+      return new Promise((resolve, reject) => {
+        const valid = tv4.validateMultiple(json, profile)
+        if (valid) {
+          resolve(valid)
+        }
+        reject(Error(tv4.error))
```

Comment:
we need to throw array also here - otherwise how user could use it if the same function could throw two different structures?